runtime/command-metrics/src/main/java/io/aklivity/zilla/runtime/command/metrics/internal/airline/ZillaMetricsCommand.java

```diff
@@ -73,44 +61,47 @@ public final class ZillaMetricsCommand extends ZillaCommand
     @Arguments(title = {"name"})
     public List<String> args;
 
-    public ZillaMetricsCommand()
-    {
-        Path engineDirectory = engineDirectory();
-        this.labelsPath = engineDirectory;
-        this.metricsPath = engineDirectory.resolve("metrics");
-    }
-
     @Override
     public void run()
     {
         String binding = args != null && args.size() >= 1 ? args.get(0) : null;
-        Map<Metric.Kind, List<MetricsLayout>> layouts = Map.of();
-        try
+        Engine engine = engine();
+        requireNonNull(engine);
+        MetricsReader metrics = new MetricsReader(engine, engine::supplyLocalName);
+        LongPredicate filterPredicate = filterPredicate(namespace, binding, engine::supplyLabelId);
+        List<MetricRecord> records = filter(metrics.records(), filterPredicate);
+        MetricsPrinter printer = new MetricsPrinter(records);
+        do
         {
-            layouts = Map.of(
-                    COUNTER, countersLayouts(),
-                    GAUGE, gaugesLayouts(),
-                    HISTOGRAM, histogramsLayouts());
-            final LabelManager labels = new LabelManager(labelsPath);
-            MetricsProcessor metrics = new MetricsProcessor(layouts, labels, namespace, binding);
-            do
-            {
-                metrics.print(System.out);
-                Thread.sleep(interval * 1000L);
-            } while (interval != 0);
-        }
-        catch (InterruptedException | IOException ex)
+            printer.print(System.out);
+            sleep(interval);
+        } while (interval != 0);
+        close(engine);
+    }
+
+    private Engine engine()
+    {
+        final Configuration config = engineConfiguration();
```

Comment:
Please inline `engineConfiguration` here, we don't need it moved out to another method.
To handle the implicit `engine.close()` via `try-with-resources`, also inline `engine()` method into `run()` too.

Reply:
yep, got it; done